File: shared/types/models_meta.ts
// Mirrors backend/api/schemas/models_meta.py

export interface ModelMetadataResponse {
  name: string;
  display_name: string;
  category: "diffusion" | "jump" | "stochastic_vol" | "empirical" | "regime";
  supports_regimes: boolean;
  is_implemented: boolean;
  description: string;
}

export interface ModelsListResponse {
  models: ModelMetadataResponse[];
}

File: backend/api/schemas/models_meta.py
from __future__ import annotations

from pydantic import BaseModel


class ModelMetadataResponse(BaseModel):
    name: str
    display_name: str
    category: str
    supports_regimes: bool
    is_implemented: bool
    description: str


class ModelsListResponse(BaseModel):
    models: list[ModelMetadataResponse]
